### DISCUSSION TITLE
useInViewport中的demo

### DISCUSSION THREAD

**Discussion by JeromeD3** [alibaba/hooks#2292] [General]
如果threshold是数组的话，由于deps这里是浅比较，会在useEffectWithTarget中引起effect()的重新执行。这种情况下不考虑优化一下么？还是说这点性能可以忽略不计。🤔小白求解答

**Comment by JeromeD3** [comment]
我觉得可以给threshold套一个useRef，但是整体代码看起来不太美观

**Comment by liuyib** [comment]
类似于 useInViewport 中使用的 [useEffectWithTarget](https://github.com/alibaba/hooks/blob/master/packages/hooks/src/utils/useEffectWithTarget.ts) 这个东西，也就是 [createEffectWithTarget](https://github.com/alibaba/hooks/blob/master/packages/hooks/src/utils/createEffectWithTarget.ts)，其内部使用了 [depsAreSame](https://github.com/alibaba/hooks/blob/master/packages/hooks/src/utils/createEffectWithTarget.ts#L41-L44) 对比依赖，depsAreSame 算法如下：

https://github.com/alibaba/hooks/blob/2ff3ed9e1b6923fecd2db98fb2af33a9656cfadf/packages/hooks/src/utils/depsAreSame.ts#L1-L9

可以看到，对于数组类型，做了一层深比较。所以，直接写数组没有性能问题
